### ISSUE TITLE
Timezone Errors with post dates

### ISSUE THREAD

**Issue by asecondwill** [timber/timber#2737]
### Expected Behavior

I'm expecting the timezone setting in wp-admin to be used by the timber filters to output the  time_ago

### Actual behavior

My site is set to Australia/Adilade
 {{ post.date('j F Y H:i') }}      shows the published date in the correct timezone.  

Time Ago
`{{ post.post_date|time_ago }}`
Time ago says "In about 7 hours" at 1230 UTC when I'm testing, again the publish date is 1930 today (16th May) - Adilaide time, which was 3 hours ago.  

UPDATE:  if i try 
` {{ post.date|time_ago }}    `

I get 14hours ago.  which is still wrong, it's 2230 in Adilade so 3 hours ago. 

### Steps to reproduce behavior

Time ago
`{{ post.post_date|time_ago }}`

with composer twig: 
 `       "timber/timber": "2.0.0.beta-1",`

### Notes

I got the same behaviour in Timber 1, so upgraded thinking this had been fixed in 2 having read through the tickets here.

### What version of Timber are you using?

2.0.0.beta-1

### What version of WordPress are you using?

Wordpress 6.2

### What version of PHP are you using?

8.0

### How did you install Timber?

Installed or updated Timber through Composer

**Comment by asecondwill** [comment]
Updated as I figured out the published date is

` {{ post.date('j F Y H:i') }}     `

not 
` {{ post.post_date|date('j F Y')}}`

**Comment by gchtr** [comment]
Most of the date functionality was fixed in 2.x, but it could very well be that I missed the `|time_ago()` filter. We will first have to add a test to confirm this is still an issue.

**Comment by Levdbas** [comment]
Did some local tests @gchtr and I recreated the same two outcomes that @asecondwill describes be it with different dates and time zones.

{{ post.post_date|time_ago }} does not seem to take the timezone into account because it relies on `time()` instead of using `current_time()` [see WP docs](https://developer.wordpress.org/reference/functions/current_time/)

{{ post.date }} as stated in the [Timber docs](https://timber.github.io/docs/reference/timber-post/#date) only returns the publication date without the time. So when using |time_ago on that it will always count use that day at midnight. That in combination with our first issue gives leads to inexplicable behavior. Using `{{post.date('j F Y h:i')|time_ago}}`, so explicitly force hours and minutes in, fixes this issue as well. So when you want to use |time_ago, it's best to use post_date.

Now I understand why i struggled so much with dates in the past as well.

**Comment by gchtr** [comment]
I found out what the issues was and added a fix in #2758.

> {{ post.date }} as stated in the [Timber docs](https://timber.github.io/docs/reference/timber-post/#date) only returns the publication date without the time. So when using |time_ago on that it will always count use that day at midnight. That in combination with our first issue gives leads to inexplicable behavior. Using `{{post.date('j F Y h:i')|time_ago}}`, so explicitly force hours and minutes in, fixes this issue as well.

Correct. You should use a full date and time string when using `{{ post.date }}`. But you don’t necessarily have to use `j F Y H:i:s`. You can also use one of the following:

```twig
{{ post.date('U')|time_ago }}
{{ post.date(constant('DATE_ATOM'))|time_ago }}
```

- `U` is a timestamp
- `DATE_ATOM` is an example for using a predefined timezone constant. See [The DateTimeInterface interface](https://www.php.net/manual/en/class.datetimeinterface.php#datetimeinterface.constants.atom). You could also use `DATE_W3C` or `DATE_RFC3339`. I just feel like `DATE_ATOM` is the easiest to remember.

> So when you want to use |time_ago, it's best to use post_date.

You can use `{{ post.post_date|time_ago }}`. That should work just as fine for the `|time_ago` filter. The only difference to using `{{ post.date('U')|time_ago }}` is that no filtering will be applied.

> {{ post.post_date|time_ago }} does not seem to take the timezone into account because it relies on `time()` instead of using `current_time()`.

Just a note here. As of WordPress 5.3, it’s not recommended to use `current_time()` anymore when you want the current time. See [Date/Time component improvements in WordPress 5.3](https://make.wordpress.org/core/2019/09/23/date-time-improvements-wp-5-3/). The correct usage to work with time in Timber is document in the [Date/Time Guide](https://timber.github.io/docs/v2/guides/date-time/).

**Comment by gchtr** [comment]
@alessandro-newzoo My suggestions don’t work yet. But they should as soon as the pull request https://github.com/timber/timber/pull/2758 is merged.

Maybe you can check it out when you test the fix by installing from the pull request’s branch?

```bash
composer require timber/timber:dev-2.x-time-ago
```

**Comment by alessandro-newzoo** [comment]
@gchtr ohhh sorry about that, I misinterpreted your previous comment!
I don't have a lot of time this week to test another version, but I might give it a shot next week.

For now, for some reason, `{{ post.date(constant('DATE_ATOM'))|time_ago }}` seems to work perfectly, so I'll use that for the time being.

Thank you SO much both for the PR and for providing a snippet that finally made me stop scratching my head after 2 days haha!

**Comment by gchtr** [comment]
@alessandro-newzoo That’s alright. I figured the first line in my comment can easily missed :).

> For now, for some reason, `{{ post.date(constant('DATE_ATOM'))|time_ago }}` seems to work perfectly, so I'll use that for the time being.

I think that works because the `DATE_ATOM` format includes the timezone in the returned date, so the underlying `strtotime()` can interpret it correctly when you use `|time_ago`.

> Thank you SO much both for the PR and for providing a snippet that finally made me stop scratching my head after 2 days haha!

You’re welcome! Dates and time made me scratch my head a lot in the past. I took quite some time to eventually "get it".

**Comment by gchtr** [comment]
Fixed in #2758.
